src/js/components/Carousel/Carousel.js

```diff
@@ -1,242 +1,218 @@
-        activeIndex: activeChild,
-        priorActiveIndex: activeIndex,
-      });
-    }
-  }
-
-  const onChildChange = useCallback(
+  const onPrevious = useCallback(() => {
+    if (inTransition) return;
+    setDirection('previous');
+    setInTransition(true);
+    setPrevious(current);
+    const next = current === 0 ? numSlides - 1 : current - 1;
+    setCurrent(next);
+    onChild(next);
+  }, [current, numSlides, inTransition, onChild]);
+
+  const onNext = useCallback(() => {
+    if (inTransition) return;
+    setDirection('next');
+    setInTransition(true);
+    setPrevious(current);
+    const next = current === numSlides - 1 ? 0 : current + 1;
+    setCurrent(next);
+    onChild(next);
+  }, [current, numSlides, inTransition, onChild]);
+
+  const onSelectorNavigation = useCallback(
     (index) => {
-      if (onChild) {
-        onChild(index);
-      }
+      if (current === index) return;
+      if (inTransition) return;
+      setDirection(index > current ? 'next' : 'previous');
+      setInTransition(true);
+      setPrevious(current);
+      setCurrent(index);
+      onChild(index);
     },
-    [onChild],
+    [current, inTransition, onChild],
   );
 
+  const onControlledNavigation = useCallback(() => {
+    if (inTransition) return;
+    if (activeChild === activeChildState) return;
+    if (activeChild === current) return;
+    setDirection(activeChild > current ? 'next' : 'previous');
+    setInTransition(true);
+    setPrevious(current);
+    setCurrent(activeChild);
+    setActiveChildState(activeChild);
+    onChild(activeChild);
+  }, [activeChild, activeChildState, current, inTransition, onChild]);
+
+  /**
+   * Handles when the "fill" or "height" & "width" props are not specified
+   * and no Carousel container is allocated for the slides. On a transition,
+   * this calculates the current dimensions of the slide and sets those as the
+   * container height and width of the Carousel.
+   */
+
   useEffect(() => {
-    if (play) {
-      const timer = setInterval(() => {
-        if (activeIndex < lastIndex) {
-          setIndexes({
-            activeIndex: activeIndex + 1,
-            priorActiveIndex: activeIndex,
-          });
-          onChildChange(activeIndex + 1);
-        } else {
-          setIndexes({
-            activeIndex: 0,
-            priorActiveIndex: activeIndex,
+    if (noDimensions) {
+      if (inTransition) {
+        const { offsetWidth: currentWidth, offsetHeight: currentHeight } =
+          containerRef.current;
+        if (currentHeight && currentWidth) {
+          setContainerProps({
+            containerHeight: currentHeight,
+            containerWidth: currentWidth,
           });
-          onChildChange(0);
         }
-      }, play);
-
-      timerRef.current = timer;
-
-      return () => {
-        clearTimeout(timer);
-      };
+      } else {
+        setContainerProps({
+          containerHeight: undefined,
+          containerWidth: undefined,
+        });
```

Comment:
This is causing the flash I see in chrome on windows at the end of a transition.

It looks visibly like the container ends up bigger briefly and the child is drawn much farther left briefly and then the container goes back to the expected size and the child ends up back where it was before.

Is this really needed?